docs/textsearch.adoc

```diff
@@ -182,4 +182,18 @@ mgmt.buildIndex('name', Vertex.class).addKey(string, Mapping.STRING.asParameter(
 mgmt.buildIndex('document', Vertex.class).addKey(text, Mapping.TEXT.asParameter()).buildMixedIndex("search")
 mgmt.commit()
 
-With these settings, JanusGraph will use a SimpleAnalyzer analyzer for property key 'string', an EnglishAnalyzer analyzer for property key 'text',  a KeywordAnalyzer analyzer for property 'name' and a StandardAnalyzer analyzer for property 'document'.
\ No newline at end of file
+With these settings, JanusGraph will use a SimpleAnalyzer analyzer for property key 'string', an EnglishAnalyzer analyzer for property key 'text',  a KeywordAnalyzer analyzer for property 'name' and a StandardAnalyzer analyzer for property 'document'.
+
+==== Custom parameters
+
+Some index backends are supporting custom parameters (right now only Elasticsearch supports custom parameters). The name of the custom parameter must be set through ParameterType.customParameterName("yourProperty").
+
+===== For Elasticsearch
+
+[source, gremlin]
+mgmt = graph.openManagement()
+myProperty = mgmt.makePropertyKey('my_property').dataType(String.class).make()
+mgmt.buildIndex('custom_property_test', Vertex.class).addKey(string, Mapping.TEXT.asParameter(), Parameter.of(ParameterType.customParameterName.("boost"), 5.5), Parameter.of(ParameterType.customParameterName.("similarity"), "DFR")).buildMixedIndex("search")
+mgmt.commit()
+
+With these settings, JanusGraph will use the boost 5.5 and DFR similarity algorithm for property key 'my_property'.
```

Comment:
I don't know much about Elasticsearch but is there some resource in their docs that we could link to here that lists other possibilities for custom parameters?

Reply:
@FlorianHockmann Thank you for your review! Mapping properties depend on the version of ElasticSearch. Is it OK to add this link for mapping properties (current ElasticSearch version)?: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-params.html

Also, it is possible to use a concrete link to specific ElasticSearch version like: https://www.elastic.co/guide/en/elasticsearch/reference/6.4/mapping-params.html

Reply:
I guess linking to a specific version like you now did is preferable as otherwise the link could break with every ElasticSearch release and we probably won't notice that.